src/main/java/io/lettuce/core/json/UnproccessedJsonValue.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2024, Redis Ltd. and Contributors
+ * All rights reserved.
+ *
+ * Licensed under the MIT License.
+ *
+ * This file contains contributions from third-party contributors
+ * licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.lettuce.core.json;
+
+import io.lettuce.core.codec.RedisCodec;
+
+import java.nio.ByteBuffer;
+
+class UnproccessedJsonValue<K, V> implements JsonValue<K, V> {
+
+    private JsonValue<K, V> jsonValue;
+
+    private final JsonParser<K, V> parser;
+
+    private boolean deserialized = false;
+
+    private final ByteBuffer unprocessedData;
+
+    private final RedisCodec<K, V> codec;
+
+    UnproccessedJsonValue(ByteBuffer bytes, RedisCodec<K, V> theCodec, JsonParser<K, V> theParser) {
+        unprocessedData = bytes;
+        parser = theParser;
+        codec = theCodec;
+    }
+
+    @Override
+    public V toValue() {
+        if (deserialized) {
+            return jsonValue.toValue();
+        }
+
+        // if no deserialization took place, so no modification took place
+        // in this case we can decode the source data
```

Comment:
If by "as is" you mean as a byte array then this would not consider the default codec, associated with the given connection and IMHO this would not be the expected behavior by the user of the driver. Typically all data received is treated as either `key` or `value` and based on that is transformed in away that the respective Codec demands to the type the Codec maps to.

For example if the codec did some compression or encryption and then transformation to String (for example) then that would mean we would get the actual String result here. Getting the compressed and/or encrypted string in raw format does not seem very meaningful to me.